okhttp-logging-interceptor/src/test/java/okhttp3/logging/HttpLoggingInterceptorTest.java

```diff
@@ -743,6 +748,61 @@ private void bodyGetNoBody(int code) throws IOException {
         .assertNoMoreLogs();
   }
 
+  @Test public void bodyLogMaxMinimumValue() throws Exception {
+    try {
+      setBodyLogMax(-2);
+      fail();
+    } catch (IllegalArgumentException e) {
+      assertEquals("bodyLogMax should be over -1", e.getMessage());
+      assertEquals(HttpLoggingInterceptor.LOG_LIMITATION_NONE, applicationInterceptor.getBodyLogMax());
+      assertEquals(HttpLoggingInterceptor.LOG_LIMITATION_NONE, networkInterceptor.getBodyLogMax());
+    }
+  }
+
+  @Test public void setBodyLogMax() throws Exception {
+    setBodyLogMax(5000);
+    assertEquals(5000, applicationInterceptor.getBodyLogMax());
+    assertEquals(5000, networkInterceptor.getBodyLogMax());
+  }
+
+  @Test public void logBodyLimitation() throws Exception {
```

Comment:
Ideally this test would verify that only the right amount is read into the buffer instead of just based on the output.  But not sure how easy that is.